pkg/controllers/workgenerator/controller.go

```diff
@@ -382,6 +385,32 @@ func (r *Reconciler) listAllWorksAssociated(ctx context.Context, resourceBinding
 func (r *Reconciler) syncAllWork(ctx context.Context, resourceBinding *fleetv1beta1.ClusterResourceBinding, existingWorks map[string]*fleetv1beta1.Work, cluster *clusterv1beta1.MemberCluster) (bool, bool, error) {
 	updateAny := atomic.NewBool(false)
 	resourceBindingRef := klog.KObj(resourceBinding)
+
+	// Refresh the apply strategy for all existing works.
+	//
+	// This step is performed separately from other refreshes as apply strategy changes are
+	// CRP-scoped and independent from the resource snapshot management mechanism. In other
+	// words, even if a work has become stranded (i.e., it is linked to a resource snapshot that
+	// is no longer present in the system), it should still be able to receive the latest apply
+	// strategy update.
+	errs, cctx := errgroup.WithContext(ctx)
+	for workName := range existingWorks {
+		w := existingWorks[workName]
+		errs.Go(func() error {
+			updated, err := r.syncApplyStrategy(ctx, resourceBinding, w)
+			if err != nil {
+				return err
+			}
+			if updated {
```

Comment:
this will revert the applied condition on the binding. It seems that it's fine but please think through.

Reply:
Hi Ryan! It might be (if a custom owner has been added before).